### DISCUSSION TITLE
pip3 install -r requirements.txt failing

### DISCUSSION THREAD

**Discussion by LycanHD** [ryelogheat/xpbot#32] [Q&A]
PS F:\> cd xpbot

PS F:\xpbot> pip3 install -r requirements.txt

Collecting pymediainfo~=4.2.1

  Using cached pymediainfo-4.2.1-py2.py3-none-win_amd64.whl (2.6 MB)

Collecting python-dotenv==0.15.0

  Using cached python_dotenv-0.15.0-py2.py3-none-any.whl (18 kB)

Requirement already satisfied: requests~=2.24.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 3)) (2.24.0)

Collecting ffmpy~=0.2.3

  Using cached ffmpy-0.2.3-py3-none-any.whl

Requirement already satisfied: guessit~=3.3.1 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 5)) (3.3.1)

Requirement already satisfied: rich~=9.10.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 6)) (9.10.0)

Requirement already satisfied: fuzzywuzzy~=0.18.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 7)) (0.18.0)

Requirement already satisfied: pyimgbox~=1.0.3 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 8)) (1.0.5)

Requirement already satisfied: torf~=3.1.3 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from -r requirements.txt (line 9)) (3.1.3)

Collecting python-Levenshtein~=0.12.2

  Using cached python-Levenshtein-0.12.2.tar.gz (50 kB)

Requirement already satisfied: setuptools in c:\program files\windowsapps\pythonsoftwarefoundation.python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\site-packages (from pymediainfo~=4.2.1->-r requirements.txt (line 1)) (56.0.0)

Requirement already satisfied: idna&lt;3,&gt;=2.5 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from requests~=2.24.0-&gt;-r requirements.txt (line 3)) (2.10)

Requirement already satisfied: chardet&lt;4,&gt;=3.0.2 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from requests~=2.24.0-&gt;-r requirements.txt (line 3)) (3.0.4)

Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,&lt;1.26,&gt;=1.21.1 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from requests~=2.24.0-&gt;-r requirements.txt (line 3)) (1.25.11)

Requirement already satisfied: certifi&gt;=2017.4.17 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from requests~=2.24.0-&gt;-r requirements.txt (line 3)) (2021.5.30)

Requirement already satisfied: babelfish in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from guessit~=3.3.1-&gt;-r requirements.txt (line 5)) (0.6.0)

Requirement already satisfied: python-dateutil in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from guessit~=3.3.1-&gt;-r requirements.txt (line 5)) (2.8.2)

Requirement already satisfied: rebulk&gt;=3 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from guessit~=3.3.1-&gt;-r requirements.txt (line 5)) (3.0.1)

Requirement already satisfied: colorama&lt;0.5.0,&gt;=0.4.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from rich~=9.10.0-&gt;-r requirements.txt (line 6)) (0.4.4)

Requirement already satisfied: pygments&lt;3.0.0,&gt;=2.6.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from rich~=9.10.0-&gt;-r requirements.txt (line 6)) (2.10.0)

Requirement already satisfied: commonmark&lt;0.10.0,&gt;=0.9.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from rich~=9.10.0-&gt;-r requirements.txt (line 6)) (0.9.1)

Requirement already satisfied: typing-extensions&lt;4.0.0,&gt;=3.7.4 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from rich~=9.10.0-&gt;-r requirements.txt (line 6)) (3.10.0.2)

Requirement already satisfied: beautifulsoup4 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (4.10.0)

Requirement already satisfied: httpx==0.*,&gt;=0.16.0 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (0.19.0)

Requirement already satisfied: charset-normalizer in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (2.0.6)

Requirement already satisfied: sniffio in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (1.2.0)

Requirement already satisfied: rfc3986[idna2008]&lt;2,&gt;=1.3 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (1.5.0)

Requirement already satisfied: httpcore&lt;0.14.0,&gt;=0.13.3 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (0.13.7)

Requirement already satisfied: flatbencode==0.2.* in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from torf~=3.1.3-&gt;-r requirements.txt (line 9)) (0.2.1)

Requirement already satisfied: h11&lt;0.13,&gt;=0.11 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpcore&lt;0.14.0,&gt;=0.13.3-&gt;httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (0.12.0)

Requirement already satisfied: anyio==3.* in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from httpcore&lt;0.14.0,&gt;=0.13.3-&gt;httpx==0.*,&gt;=0.16.0-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (3.3.1)

Requirement already satisfied: soupsieve&gt;1.2 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from beautifulsoup4-&gt;pyimgbox~=1.0.3-&gt;-r requirements.txt (line 8)) (2.2.1)

Requirement already satisfied: six&gt;=1.5 in c:\users\gemini\appdata\local\packages\pythonsoftwarefoundation.python.3.8_qbz5n2kfra8p0\localcache\local-packages\python38\site-packages (from python-dateutil-&gt;guessit~=3.3.1-&gt;-r requirements.txt (line 5)) (1.16.0)

Building wheels for collected packages: python-Levenshtein

  Building wheel for python-Levenshtein (setup.py) ... error

  ERROR: Command errored out with exit status 1:

   command: 'C:\Users\Gemini\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"'; __file__='"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec&#40;compile(code, __file__, '"'"'exec'"'"'&#41;)' bdist_wheel -d 'C:\Users\Gemini\AppData\Local\Temp\pip-wheel-_7t99qcn'

       cwd: C:\Users\Gemini\AppData\Local\Temp\pip-install-oxl5ypo2\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\

  Complete output (28 lines):

  running bdist_wheel

  running build

  running build_py

  creating build

  creating build\lib.win-amd64-3.8

  creating build\lib.win-amd64-3.8\Levenshtein

  copying Levenshtein\StringMatcher.py -&gt; build\lib.win-amd64-3.8\Levenshtein

  copying Levenshtein\__init__.py -&gt; build\lib.win-amd64-3.8\Levenshtein

  running egg_info

  writing python_Levenshtein.egg-info\PKG-INFO

  writing dependency_links to python_Levenshtein.egg-info\dependency_links.txt

  writing entry points to python_Levenshtein.egg-info\entry_points.txt

  writing namespace_packages to python_Levenshtein.egg-info\namespace_packages.txt

  writing requirements to python_Levenshtein.egg-info\requires.txt

  writing top-level names to python_Levenshtein.egg-info\top_level.txt

  adding license file 'COPYING' (matched pattern 'COPYING*')

  reading manifest file 'python_Levenshtein.egg-info\SOURCES.txt'

  reading manifest template 'MANIFEST.in'

  warning: no previously-included files matching '*pyc' found anywhere in distribution

  warning: no previously-included files matching '*so' found anywhere in distribution

  warning: no previously-included files matching '.project' found anywhere in distribution

  warning: no previously-included files matching '.pydevproject' found anywhere in distribution

  writing manifest file 'python_Levenshtein.egg-info\SOURCES.txt'

  copying Levenshtein\_levenshtein.c -&gt; build\lib.win-amd64-3.8\Levenshtein

  copying Levenshtein\_levenshtein.h -&gt; build\lib.win-amd64-3.8\Levenshtein

  running build_ext

  building 'Levenshtein._levenshtein' extension

  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/

  ----------------------------------------

  ERROR: Failed building wheel for python-Levenshtein

  Running setup.py clean for python-Levenshtein

Failed to build python-Levenshtein

Installing collected packages: python-Levenshtein, python-dotenv, pymediainfo, ffmpy

    Running setup.py install for python-Levenshtein ... error

    ERROR: Command errored out with exit status 1:

     command: 'C:\Users\Gemini\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"'; __file__='"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec&#40;compile(code, __file__, '"'"'exec'"'"'&#41;)' install --record 'C:\Users\Gemini\AppData\Local\Temp\pip-record-s46k0mi1\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\Gemini\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\python-Levenshtein'

         cwd: C:\Users\Gemini\AppData\Local\Temp\pip-install-oxl5ypo2\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\

    Complete output (28 lines):

    running install

    running build

    running build_py

    creating build

    creating build\lib.win-amd64-3.8

    creating build\lib.win-amd64-3.8\Levenshtein

    copying Levenshtein\StringMatcher.py -&gt; build\lib.win-amd64-3.8\Levenshtein

    copying Levenshtein\__init__.py -&gt; build\lib.win-amd64-3.8\Levenshtein

    running egg_info

    writing python_Levenshtein.egg-info\PKG-INFO

    writing dependency_links to python_Levenshtein.egg-info\dependency_links.txt

    writing entry points to python_Levenshtein.egg-info\entry_points.txt

    writing namespace_packages to python_Levenshtein.egg-info\namespace_packages.txt

    writing requirements to python_Levenshtein.egg-info\requires.txt

    writing top-level names to python_Levenshtein.egg-info\top_level.txt

    adding license file 'COPYING' (matched pattern 'COPYING*')

    reading manifest file 'python_Levenshtein.egg-info\SOURCES.txt'

    reading manifest template 'MANIFEST.in'

    warning: no previously-included files matching '*pyc' found anywhere in distribution

    warning: no previously-included files matching '*so' found anywhere in distribution

    warning: no previously-included files matching '.project' found anywhere in distribution

    warning: no previously-included files matching '.pydevproject' found anywhere in distribution

    writing manifest file 'python_Levenshtein.egg-info\SOURCES.txt'

    copying Levenshtein\_levenshtein.c -&gt; build\lib.win-amd64-3.8\Levenshtein

    copying Levenshtein\_levenshtein.h -&gt; build\lib.win-amd64-3.8\Levenshtein

    running build_ext

    building 'Levenshtein._levenshtein' extension

    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/

    ----------------------------------------

ERROR: Command errored out with exit status 1: 'C:\Users\Gemini\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"'; __file__='"'"'C:\\Users\\Gemini\\AppData\\Local\\Temp\\pip-install-oxl5ypo2\\python-levenshtein_076f291c02bf4797b5c35f9c56a72d15\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec&#40;compile(code, __file__, '"'"'exec'"'"'&#41;)' install --record 'C:\Users\Gemini\AppData\Local\Temp\pip-record-s46k0mi1\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\Gemini\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Include\python-Levenshtein' Check the logs for full command output.

**Comment by ryelogheat** [comment]
Try using a more recent Python version. Just recently I setup the bot on a new server while using Python3.9 & didn't have any issues. 

(ik you probably either moved on or figured out already but for anyone who is still having issues, try using a newer python version & ideally use a py virtual environment to prevent any dependency issues)
